packages/generate-persisted-query-manifest/src/__tests__/cli.test.ts

```diff
@@ -302,6 +306,32 @@ query   GreetingQuery
 });
 
 test("ensures manifest bodies and id hash applies document transforms", async () => {
+  async function removeClientDirective(
+    query: DocumentNode,
+  ): Promise<DocumentNode> {
+    try {
+      // v4
+      // @ts-ignore
+      // prettier-ignore
+      const { removeDirectivesFromDocument } = await import("@apollo/client/utilities/internal");
+
+      return removeDirectivesFromDocument(
+        [{ name: "client", remove: true }],
+        query,
+      )!;
+    } catch (e) {
+      // v3
+      // @ts-ignore
+      // prettier-ignore
+      const { removeDirectivesFromDocument } = await import("@apollo/client/utilities");
+
+      return removeDirectivesFromDocument(
+        [{ name: "client", remove: true }],
+        query,
+      )!;
+    }
```

Comment:
```suggestion
    const v4Entrypoint = "@apollo/client/utilities/internal";
    const v3Entrypoint = "@apollo/client/utilities";
    type RemoveDirectivesFromDocument = (
      directives: Array<{
        name?: string;
        test?: (node: DirectiveNode) => boolean;
        remove?: boolean;
      }>,
      doc: DocumentNode,
    ) => DocumentNode | null;

    const removeDirectivesFromDocument = await import(v4Entrypoint)
      .catch(() => import(v3Entrypoint))
      .then(
        (module: {
          removeDirectivesFromDocument: RemoveDirectivesFromDocument;
        }) => module.removeDirectivesFromDocument,
      );

    return removeDirectivesFromDocument(
      [{ name: "client", remove: true }],
      query,
    )!;
```

This is not particularly shorter, but it would avoid the `@ts-expect-error` and `//prettier-ignore` calls.

But in the end, either approach is fine :)